### ISSUE TITLE
[java] UnnecessaryImport: false positive for static on-demand imports

### ISSUE THREAD

**Issue by kamesh95** [pmd/pmd#3616]
**Affects PMD Version:**  6.40.0

**Rule:** [UnnecessaryImport](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#unnecessaryimport)

**Description:**
When I analyze the sample code at https://docs.mongodb.com/drivers/java/sync/current/quick-start/ with a minor change in the filters import, i.e. using import static `com.mongodb.client.model.Filters.*;` instead of import static `com.mongodb.client.model.Filters.eq;` pmd shows unused import com.mongodb.client.model.Filters.* even if the eq operator is being used in the code. This works fine for static imports of Math class or System class. 

Additional dependencies required for mongo:

```
dependencies {
  compile 'org.mongodb:mongodb-driver-sync:4.3.4'
}
```

**Code Sample demonstrating the issue:**

```java
import static com.mongodb.client.model.Filters.*; // This is shown as a violation [Incorrect]
import static java.lang.Math.*; // No Violation [Correct]
import static java.lang.System.*; // No Violation [Correct]
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;
import org.springframework.boot.SpringApplication;


public class Auto {
    public static void main(String[] args) {
        SpringApplication.run(Auto.class, args);
        String uri = "<connection string uri>";
        try (MongoClient mongoClient = MongoClients.create(uri)) {
            MongoDatabase database = mongoClient.getDatabase("sample_mflix");
            MongoCollection<Document> collection = database.getCollection("movies");
            Document doc = collection.find(eq("title", "Back to the Future")).first(); // eq is used here from the Filters.* import at line 1
            out.println(doc.toJson());
            double var1= sqrt(5.0);
        }
    }
}

```

**Expected outcome:**
PMD should not show violation for unused import.

PMD reports a violation at line 1, but that's wrong. That's a false positive.

**Running PMD through:** *[CLI]*
pmd -d src/main/java/com/test/auto/Auto.java -f json -R category/java/codestyle.xml/UnnecessaryImport

**Comment by adangel** [comment]
Thanks for the report. The problem is, that PMD can only figure out that `eq` is indeed `com.mongodb.client.model.Filters.eq` if you tell PMD about all your classes using the [`--aux-classpath`](https://pmd.github.io/latest/pmd_userdocs_cli_reference.html#-aux-classpath) option. Since this is usually a bit more complex than for simple "one file java projects", PMD is seldom used from the command line. E.g. on your project, the classpath consists of at least 3 jars already (mongodb-driver-sync, mongodb-driver-core, bson - not counting spring...). Therefore usually a PMD plugin for the build tools is used.

You seem to use gradle, so please use [The PMD Plugin](https://docs.gradle.org/current/userguide/pmd_plugin.html). This will make sure to call PMD with the proper auxclasspath.
See https://github.com/darrendanvers/pmd-gradle-example for an example.

I'm closing this issue as not reproducible.
